src/dash-table/utils/actions.js

```diff
@@ -53,23 +61,36 @@ export const clearSelection = {
     selected_cells: []
 };
 
-export function editColumnName(column, columns, headerRowIndex) {
+export function changeColumnHeader(column, columns, headerRowIndex, mergeDuplicateHeaders, newColumnName) {
+    let cloneColumn = Object.assign({}, column);
+    if ((typeof column.name === 'string') || (column.name instanceof String)) {
+            const columnHeaders = columns.map(col => col.name);
+            const maxLength = findMaxLength(columnHeaders);
+            const newColumnNames = Array(maxLength).fill(column.name);
+            cloneColumn = Object.assign({}, column, {name: newColumnNames});
+    }
+    const transformedColumns = columns.map(col => {
+        if (((typeof column.name === 'string') || (column.name instanceof String)) && col.id === column.id) {
+            return cloneColumn;
+        } else {
+            return col;
+        }
+    });
```

Comment:
Instead of a map you can:

newColumns = columns.slice(0) // shallow clone of array
newColumns[newColumns.indexOf(column)] = newColumn // replace the modified entry